hazelcast-jet-core/src/main/java/com/hazelcast/jet/Sources.java

```diff
@@ -124,11 +130,25 @@ private Sources() {
      */
     public static <K, V, T> Source<T> readMap(
             @Nonnull String mapName,
-            @Nonnull DistributedPredicate<Map.Entry<K, V>> predicate,
+            @Nonnull Predicate<K, V> predicate,
+            @Nonnull Projection<Entry<K, V>, T> projection
+    ) {
+        return fromProcessor("readMap(" + mapName + ')',
+                SourceProcessors.readMapP(mapName, predicate, projection));
+    }
+
+    /**
+     * Convenience for {@link #readMap(String, Predicate, Projection)}
+     * which use a {@link DistributedPredicate} and {@link DistributedFunction}
+     * as the predicate and projection functions.
+     */
```

Comment:
I would mention here that it precludes the usage of index, because this is an easy pitfall. I would remove the mention of index from methods with non-native predicate, as this is a basic feature of the Predicate as documented in IMGD's manual.